### PR TITLE
Add settings overview for digital watchface

### LINKED ISSUE #582: Battery Percentage Icon
Could an optional battery icon that shows the actual percentage be added, either next to the battery icon, or (preferably) the CM/LOS method, where a circular battery icon has the percentage shown within?

### PR DESCRIPTION
**Pull request by Sykomaniac**
### What does it do?
Create a settings overlay for the digital watchface to add colourful battery icon and battery percentage display. Implementation is similar to PineTime to avoid adding entries into main settings pages. Closes #582 

### Screenshots
PXL_20220310_225147961 MP
PXL_20220310_225156661
PXL_20220310_225207912
PXL_20220310_225213511

### CODE REVIEW

**Comment by kieranc** [comment]
I'm very happy to see this implemented, it was on my to-do list!
The settings icon doesn't seem to be quite centered on the screen, is this intentional?
It feels like there are a lot of options which could be implemented but finding space for all the settings could be complicated. BG/text color, seconds display, date format etc. I wonder if the settings buttons which appears on long press could become 2 buttons, leading to 2 pages, one for color settings and one for other options?

**Comment by NeroBurner** [comment]
works great with InfiniSim

InfiniSim_2022-03-13_202335 InfiniSim_2022-03-13_202337 InfiniSim_2022-03-13_202340
InfiniSim_2022-03-13_202348 InfiniSim_2022-03-13_202352 InfiniSim_2022-03-13_202409
InfiniSim_2022-03-13_202421 InfiniSim_2022-03-13_202436

File: src/displayapp/screens/WatchFaceDigital.cpp [hunk 1]
```diff
@@ -110,11 +181,19 @@ void WatchFaceDigital::Refresh() {
...
   if (batteryPercentRemaining.IsUpdated()) {
     auto batteryPercent = batteryPercentRemaining.Get();
-    if (batteryPercent == 100) {
-      lv_obj_set_style_local_text_color(batteryIcon, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_GREEN);
+
+    if (settingsController.GetDWSBatteryColorStatus() == Controllers::Settings::BatteryColor::ON) {
+        lv_obj_set_style_local_text_color(batteryIcon, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, BatteryIcon::GetBatteryColor(batteryPercent));
     } else {
-      lv_obj_set_style_local_text_color(batteryIcon, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_WHITE);
+        lv_obj_set_style_local_text_color(batteryIcon, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, BatteryIcon::GetDefaultBatteryColor(batteryPercent));
```
Comment (NeroBurner) [inline src/displayapp/screens/WatchFaceDigital.cpp:188]: here 2 extra spaces indentation slipped in.

File: src/displayapp/screens/WatchFaceDigital.cpp [hunk 2]
```diff
@@ -110,11 +181,19 @@ void WatchFaceDigital::Refresh() {
   batteryPercentRemaining = batteryController.PercentRemaining();
   if (batteryPercentRemaining.IsUpdated()) {
     auto batteryPercent = batteryPercentRemaining.Get();
-    if (batteryPercent == 100) {
-      lv_obj_set_style_local_text_color(batteryIcon, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_GREEN);
+
+    if (settingsController.GetDWSBatteryColorStatus() == Controllers::Settings::BatteryColor::ON) {
+        lv_obj_set_style_local_text_color(batteryIcon, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, BatteryIcon::GetBatteryColor(batteryPercent));
```
Comment (NeroBurner) [inline src/displayapp/screens/WatchFaceDigital.cpp:186]: here 2 extra spaces indentation slipped in.

**Review by NeroBurner — CHANGES_REQUESTED** [review]
you've changed Settings (i.e. you added new settings). You need to increment `settingsVersion`

```cpp
static constexpr uint32_t settingsVersion = 0x0003;
```

otherwise you are bound to run into the same errors as in PR https://github.com/InfiniTimeOrg/InfiniTime/pull/1024#issuecomment-1065530079

**Comment by Sykomaniac** [comment]
@NeroBurner thanks for that I've incremented the settings version number
@kieranc no you are right that was an oversight and now corrected

**Comment by mashuptwice** [comment]
As I've mentioned in #1205 , I don't quite see the necessity to add options about the battery feedback as quick settings. I cannot imagine anyone changing these settings very often, usually if someone prefers to have the battery percentage displayed in the status bar he will set it one time and keep it that way. 

Don't get me wrong, I am sure such quicksettings could be very useful for other options that someone would change multiple times a day.

I also very much like the idea of the colored battery icon!

**Comment by lman0** [comment]
any news @Sykomaniac ?

**Comment by Sykomaniac** [comment]
Happy to update the PR to the latest code base if desired

**Pull request closed without merging**